File: server/lib/sharePreview.js
/**
 * Scoped share preview DTO (F-03).
 * Anonymous preview must not expose health, vet, insurance, chip, or owner email.
 */
import { dateToIsoDate } from './calendarDate.js';

/**
 * @param {object} row pets row
 */
export function buildSharePreviewPet(row) {
  const dob = row.date_of_birth ? dateToIsoDate(row.date_of_birth) : null;
  return {
    name: row.name,
    species: row.species,
    breed: row.breed || '',
    age: row.age ?? null,
    date_of_birth: dob,
    photo_path: row.photo_path || null,
  };
}

/**
 * @param {object} row users row (owner)
 */
export function buildSharePreviewOwner(row) {
  return {
    first_name: row?.first_name || '',
  };
}

/**
 * @param {object} link pet_share_links row
 * @param {object} petRow pets row
 * @param {object} ownerRow users row
 */
export function buildSharePreviewResponse(link, petRow, ownerRow) {
  const expiresAt = link.expires_at
    ? (link.expires_at.toISOString?.() || String(link.expires_at))
    : null;
  return {
    link_status: link.status || 'pending',
    expires_at: expiresAt,
    access_role: link.access_role || 'carer',
    pet: buildSharePreviewPet(petRow),
    owner: buildSharePreviewOwner(ownerRow),
  };
}

File: flutter_app/lib/features/sharing/domain/entities/share_preview.dart
class SharePreview {
  const SharePreview({
    required this.pet,
    this.owner,
    this.linkStatus,
    this.accessRole,
    this.expiresAt,
  });

  final Map<String, dynamic> pet;
  final Map<String, dynamic>? owner;
  final String? linkStatus;
  final String? accessRole;
  final String? expiresAt;

  factory SharePreview.fromJson(Map<String, dynamic> json) {
    return SharePreview(
      pet: (json['pet'] as Map<String, dynamic>?) ?? {},
      owner: json['owner'] as Map<String, dynamic>?,
      linkStatus: json['link_status']?.toString(),
      accessRole: json['access_role']?.toString(),
      expiresAt: json['expires_at']?.toString(),
    );
  }
}
